allel/stats/roh.py

```diff
@@ -150,6 +151,115 @@ def _mhmm_predict_roh_state(model, is_het, pos, is_accessible, contig_size):
+        the population, but also on mutation rate and genotype error rate.
+    transition: float, optional
+        Probability of moving between states. This is based on windows, so a larger window size may call for a
+        commeasurately larger transitional probability
+    window_size: integer, optional
+        Window size (equally accessible bases) to consider as a potential ROH.
+        and recombination rate. Setting this window too small may result in spurious ROH calls, while too large will
+        result in a lack of resolution.
+    is_accessible: array_like, bool, shape (`contig_size`,), optional
+        Boolean array for each position in contig describing whether accessible
+        or not. Although optional, highly recommended so invariant sites are distinguishable from sites where variation
+        is inaccessible
+    contig_size: integer, optional
+        If is_accessible is not available, use this to specify the size of the contig, and assume all sites are
+        accessible.
+
+
+    Returns
+    -------
+    df_roh: DataFrame
+        Data frame where each row describes a run of homozygosity. Columns are 'start',
+        'stop', 'length' and 'is_marginal'. Start and stop are 1-based, stop-inclusive.
+    froh: float
+        Proportion of genome in a ROH.
+
+    Notes
+    -----
+    This function requires `pomegranate` (>= 0.9.0) and `intervaltree` to be installed.
+    installed.
+
+    """
+
+    from pomegranate import HiddenMarkovModel, PoissonDistribution
+    from intervaltree import Interval, IntervalTree
+
+    # equally accessbile windows
+    if is_accessible is None:
+        assert contig_size is not None, \
+            "If accessibility not provided must specify size of contig"
+        is_accessible = np.ones((contig_size,), dtype="bool")
+
+    eqw = equally_accessible_windows(is_accessible, window_size)
+
+    ishet = GenotypeVector(gv).is_het()
+    counts, wins, records = windowed_statistic(pos, ishet, np.sum, windows=eqw)
+
+    # heterozygote probabilities
+    het_px = np.concatenate([(phet_roh,), phet_nonroh])
+
+    # start probabilities (all equal)
+    start_prob = np.repeat(1/het_px.size, het_px.size)
+
+    # transition between underlying states
+    transition_mx = _hmm_derive_transition_matrix(transition, het_px.size)
+
+    dists = [PoissonDistribution(x * window_size) for x in het_px]
+
+    model = HiddenMarkovModel.from_matrix(transition_probabilities=transition_mx,
+                                          distributions=dists,
+                                          starts=start_prob)
+
+    prediction = np.array(model.predict(counts[:, None]))
+    pred_roh = np.where(prediction == 0)[0]
+
+    index_ivtree = IntervalTree.from_tuples([(x, x + 1) for x in pred_roh])
+    index_ivtree.merge_overlaps()
+
+    pos_ivtree = IntervalTree()
+    for iv in index_ivtree.items():
+        pos_ivtree.append(Interval(eqw[iv.begin, 0], eqw[iv.end - 1, 1]))
```

Comment:
Doesn't look like an interval tree is needed here? A simple list of tuples would do it? Simplest code is best.